rust/automerge/src/query/nth.rs

```diff
@@ -45,12 +35,16 @@ impl<'a> Nth<'a> {
         if let Some(e) = self.ops.first().and_then(|op| op.elemid()) {
             Ok(Key::Seq(e))
         } else {
-            Err(AutomergeError::InvalidIndex(self.target))
+            Err(AutomergeError::InvalidIndex(self.idx.target - 1))
```

Comment:
Maybe we should use `saturating_sub` here? It's technically not necessary because we initialize `ListState` with `target + 1` so I think `target` should always be `> 0`, but we're in an error branch here already so it's probably worth programming defensively to avoid the panic.